src/db.test.ts

```diff
@@ -341,6 +342,7 @@ Deno.test("transactions can be nested", function () {
     } catch (_) { /* ignore */ }
   });
 
+  assert(db.autoCommit);
```

Comment:
I think we should add `assert(!db.autoCommit)` calls inside of the transaction blocks as well.